### PR TITLE
Added text for drift differentiation

### PR DESCRIPTION
**Pull request by nirhutnik**
(For now only added paragraph in drift guide and TrainTestFeatureDrift - will add for the other relevant checks when we agree on how to phrase it).

### CODE REVIEW

File: deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ class TrainTestFeatureDrift(TrainTestCheck, ReduceMixin):
     We also support Population Stability Index (PSI).
     See https://www.lexjansen.com/wuss/2017/47_Final_Paper_PDF.pdf.
 
+    For categorical variables, it is recommended to use Cramer’s V, unless your variable includes categories with a
+    small number of samples, where PSI would perform better. However, in cases of a variable with a large number of
+    unique categories, it is still recommended to use Cramer’s V.
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py:47]: Not clear enough to me what "unique categories" are. I guess you meant "with only a single occurrence"?

Also, "include categories with a small number of samples". Any amount of such categories?

Comment (nirhutnik) [reply deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py:47]: 1. Actually not what I meant. How's now?
2. I don't feel comfortable enough to say. It is common to say that Cramer's V is less accurate when <5 samples, but it's a continuous value of course and fluctuates at higher values as well. Changed to fit this. WDYT?

**Review by noamzbr — APPROVED** [review]
Approved assuming update to the train_test_feature_drift check as well

File: docs/source/user-guide/general/drift_guide.rst [hunk 2]
```diff
@@ -101,12 +101,26 @@ There are many measures that can be used for this, such as the Kolmogorov-Smirno
...
 These methods have the advantage of being simple to use and produce explainable results. However, they are limited by
 checking each feature one at a time, and cannot detect drift in the relations between features. Also, these methods
 will usually detect drift multiple times if it occurs in several features.
 
+Choosing the Correct Method to Detect Drift
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+As mentioned above, we recommend to use either `Cramer's V <https://en.wikipedia.org/wiki/Cram%C3%A9r%27s_V>`__ or
+`PSI <https://www.lexjansen.com/wuss/2017/47_Final_Paper_PDF.pdf>`__ for categorical variables, and use Cramer's V by default.
+PSI is widely used in the industry, but does not have an upper limit and is not very explainable.
+Cramer's V is always in the range [0,1], and can be interpreted as the correlation between the variable's distribution and the dataset (train or test).
```
Comment (shir22) [inline docs/source/user-guide/general/drift_guide.rst:115]: last part isn't clear

File: docs/source/user-guide/general/drift_guide.rst [hunk 3]
```diff
@@ -101,12 +101,26 @@ There are many measures that can be used for this, such as the Kolmogorov-Smirno
...
+
+Both methods are very sensitive to changes in small categories. However, in deepchecks we implement a correction on PSI that prevents the value from diverging.
```
Comment (shir22) [inline docs/source/user-guide/general/drift_guide.rst:117]:
```suggestion
Both methods are very sensitive to changes in small categories. However, in deepchecks we've implemented a correction to PSI that prevents the value from diverging.
```

File: docs/source/user-guide/general/drift_guide.rst [hunk 4]
```diff
@@ -101,12 +101,26 @@ There are many measures that can be used for this, such as the Kolmogorov-Smirno
...
+
+In general, it is recommended to use Cramer's V, unless your variable includes categories with a small number of samples (common practice is under 5 samples).
```
Comment (shir22) [inline docs/source/user-guide/general/drift_guide.rst:119]:
```suggestion
In general, it is recommended to use Cramer's V, unless your variable includes categories with a small number of samples (common practice is categories with less than 5 samples).
```

File: docs/source/user-guide/general/drift_guide.rst [hunk 5]
```diff
@@ -101,12 +101,26 @@ There are many measures that can be used for this, such as the Kolmogorov-Smirno
...
+However, in cases of a variable with a large number of categories with few samples, it is still recommended to use Cramer's V, as PSI will not be able to detect change in the smaller categories.
```
Comment (shir22) [inline docs/source/user-guide/general/drift_guide.rst:120]:
```suggestion
However, in cases of a variable with many categories with few samples, it is still recommended to use Cramer's V, as PSI will not be able to detect change in the smaller categories.
```

**Review by shir22 — APPROVED** [review]

**Pull request merged**